compiler/optimizer/LoopVersioner.cpp

```diff
@@ -192,7 +192,7 @@ int32_t TR_LoopVersioner::performWithDominators()
       {
          printf("WARNING: method may have infinite loops\n");
       }
-
+   _postDominators = NULL;
```

Comment:
This is incorrect. `performWithoutDominators` calls `detectChecksToBeEliminated` which then makes use of `_postDominators`. So setting the field to `NULL` here alters the semantics which is not what we want.

`performWithoutDominators` is also ill-named. If one of it's callees makes use of post dominators then it is pretty misleading.

Reply:
ah dang that's right. That name `performWithoutDominators` threw me off. When I came across that method I didn't think it would be called from `performWithoutDominators`. But point noted for future, don't trust the method names.

As for this change, instead of setting the filed to `NULL` I could move the allocation to heap, that would also prevent scope issues in future.

Reply:
How about:

```
...
auto result = performWithoutDominators();

// Local variable postDominators is about to go out of scope. NULL this field to prevent illegal access to a stack allocated object
_postDominators = NULL;

return result;
```

Reply:
I have updated the commit